Sources/XcbeautifyLib/JunitReporter.swift

```diff
@@ -21,12 +21,16 @@ package final class JunitReporter {
 
         if FailingTestCaptureGroup.regex.match(string: line) {
             guard let testCase = generateFailingTest(line: line) else { return }
+            // reduce failing retrys, if any
+            components.removePreviousFailingTestsAfter(testCase)
             components.append(.failingTest(testCase))
         } else if RestartingTestCaptureGroup.regex.match(string: line) {
             guard let testCase = generateRestartingTest(line: line) else { return }
             components.append(.failingTest(testCase))
         } else if TestCasePassedCaptureGroup.regex.match(string: line) {
             guard let testCase = generatePassingTest(line: line) else { return }
+            // filter out failing retrys, if any
+            components.removePreviousFailingTestsAfter(testCase)
```

Comment:
If a developer enables the "Run tests repeatedly" option, and each iteration passes, this will only show as 1 success, right? I don't think that's the desired behavior, and I don't think the current proposed changes consider that scenario. It's not one covered by the existing unit tests, but it should be.

Reply:
Good point, pushed some changes, thanks!